packages/waku/src/router/fs-router.ts

```diff
@@ -22,19 +14,25 @@ const removeStart = (s: string) => {
 };
 
 export function unstable_fsRouter(
-  importMetaUrl: string,
-  loadPage: (file: string) => Promise<any> | undefined,
+  /**
+   * A mapping from a file path to a route module, e.g.
+   *   {
+   *     "_layout.tsx": () => ({ default: ... }),
+   *     "index.tsx": () => ({ default: ... }),
+   *     "foo/index.tsx": () => ...,
+   *   }
+   * This mapping can be created by Vite's import.meta.glob, e.g.
+   *   import.meta.glob("/src/pages/**\/*.tsx", { base: "/src/pages" })
+   */
+  pages: { [file: string]: () => Promise<any> },
```

Comment:
I can probably accept this change, but now it looks more dependent on vite behavior than before.

I wonder if we can expose some low-level API from Waku's minimal API to hide/isolate Vite dependency.

We still have some time, so let's consider other possibilities. (It can be within this PR, or perhaps separate from this PR.)

Reply:
> now it looks more dependent on vite behavior than before.

Implementing `import.meta.glob` is just a syntax sugar of

```js
{
  './index.tsx': () => import("/src/pages/index.tsx")
  ...
}
```

and I didn't see this specifically manifests "Vite dependency". Can you explain how this looks more Vite dependent? Did previous approach make it easier to not rely on `import.meta.glob`?

Technically, compared to previous approach, which assumed file system to have the exact structure, now it looks more flexible since users can specify import or "route module" directly in the map, such as:

```js
{
  'index': () => import("/src/pages/anywhere-user-want.tsx"),
  'foo' () => ({ default: function FooDirectlyHere() { ... }),
  ...
}
```

Reply:
If it's only about "API shape", then there's an alternative such as:

```js
function fsRouter(
  files: string[],
  loadPage: (file: string) => Promise<any>,
  ...
) { ... }
```

The requirement is technically we want to move out `files` collection to be done outside.

Reply:
> The requirement is that we want to move out files collection to be done outside.

Understandable. We can look for alternative ideas separately after merging this PR.